src/transformState.js

```diff
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const impact of actions) {
+    switch (impact.type) {
+      case 'addProperties':
+        Object.assign(state, impact['extraData']);
+        break;
+
+      case 'removeProperties':
+        for (const key of impact['keysToRemove']) {
+          delete state[key];
+        }
+        break;
+
+      case 'clear':
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
+
+      default:
+        break;
```

Comment:
you could just make `clear` path a default behaviour:
```suggestion
      default:
        for (const key in state) {
          delete state[key];
        }
```